lib/trie/proof_test.go

```diff
@@ -0,0 +1,143 @@
+package trie
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	crand "crypto/rand"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/require"
+)
+
+func inMemoryChainDB(t *testing.T) (*chaindb.BadgerDB, func()) {
+	t.Helper()
+
+	tmpdir, err := ioutil.TempDir("", "trie-chaindb-*")
+	require.NoError(t, err)
+
+	db, err := chaindb.NewBadgerDB(&chaindb.Config{
+		InMemory: true,
+		DataDir:  tmpdir,
+	})
+	require.NoError(t, err)
+
+	clear := func() {
+		err := db.ClearAll()
+		require.NoError(t, err)
+
+		err = db.Close()
+		require.NoError(t, err)
+
+		err = os.RemoveAll(tmpdir)
+		require.NoError(t, err)
+	}
+
+	return db, clear
+}
+
+func TestVerifyProof(t *testing.T) {
+	trie, entries := randomTrie(t, 500)
+	root, err := trie.Hash()
+	require.NoError(t, err)
+
+	for _, kv := range entries {
+		t.Run("", func(t *testing.T) {
```

Comment:
We do not require subtests since we are testing the same thing. Remove this line.

Reply:
Update